### PR TITLE
Add fullscreen mode to web viewer

### PR DESCRIPTION
**Pull request by jprochazk**
### What

- Closes https://github.com/rerun-io/rerun/issues/6433

This PR enables the `ToggleFullscreen` UI command for web target, implemented in JS land. The functionality is also exposed via the `toggle_fullscreen` method. Callbacks are used to communicate about fullscreen state between WASM and JS.

To test, you'll need to build branch `fullscreen-temp` of https://github.com/rerun-io/web-viewer-example with a globally linked `web-viewer` package built from source:

1. Install Yarn: `npm i -g yarn`
2. In this repository:
   - `yarn --cwd rerun_js install`
   - `yarn --cwd rerun_js/web-viewer build`
   - `yarn --cwd rerun_js/web-viewer link`
3. In the `web-viewer-example` repository:
   - `yarn`
   - `yarn link @rerun-io/web-viewer`
   - `yarn dev` + open the page

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6461?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6461?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6461)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: crates/re_viewer/src/app.rs [hunk 1]
```diff
@@ -1325,6 +1334,56 @@ impl App {
             1.0
         }
     }
+
+    #[allow(dead_code)] // only used in web
```
Comment (emilk) [inline crates/re_viewer/src/app.rs:1338]: Really? But it contains `#[cfg(not(target_arch = "wasm32"))]` code?

File: crates/re_viewer/src/app.rs [hunk 2]
```diff
@@ -1325,6 +1334,56 @@ impl App {
...
+            let egui_ctx = &self.re_ui.egui_ctx;
+            let fullscreen = egui_ctx.input(|i| i.viewport().fullscreen.unwrap_or(false));
+            egui_ctx.send_viewport_cmd(egui::ViewportCommand::Fullscreen(!fullscreen));
+        }
+
+        #[cfg(target_arch = "wasm32")]
+        {
+            if let Some(options) = &self.startup_options.fullscreen_options {
+                // Tell JS to toggle fullscreen.
+                options.on_toggle.call().unwrap();
```
Comment (emilk) [inline crates/re_viewer/src/app.rs:1351]: Let's replace the unwraps with a log call using one of the helpers in `crates/re_log/src/result_extensions.rs`

File: crates/re_viewer/src/app.rs [hunk 3]
```diff
@@ -1325,6 +1334,56 @@ impl App {
...
+        {
+            let egui_ctx = &self.re_ui.egui_ctx;
+            return egui_ctx.input(|i| i.viewport().fullscreen.unwrap_or(false));
+        }
+
+        #[cfg(target_arch = "wasm32")]
+        {
+            if let Some(options) = &self.startup_options.fullscreen_options {
+                // Ask JS if fullscreen is on or not.
+                return options.get_state.call().unwrap().is_truthy();
```
Comment (emilk) [inline crates/re_viewer/src/app.rs:1381]: Same here

**Review by emilk — APPROVED** [review]

File: crates/re_viewer/src/app.rs (hunk 2 above)
Comment (jprochazk) [reply crates/re_viewer/src/app.rs]: couldn't use those because `JsValue` does not implement `Display` - used the web tools `string_from_js_value` instead

**Comment by jprochazk** [comment]
link checker failure is unrelated

**Pull request merged**